tests/components/minecraft_server/test_init.py

```diff
@@ -108,115 +115,192 @@ def create_v1_mock_binary_sensor_entity_entry(
+
+    with patch(
+        "mcstatus.server.JavaServer.lookup",
+        return_value=JavaServer(host=TEST_HOST, port=TEST_PORT),
+    ), patch(
+        "mcstatus.server.JavaServer.async_status",
+        return_value=TEST_JAVA_STATUS_RESPONSE,
+    ):
+        assert await hass.config_entries.async_setup(java_mock_config_entry.entry_id)
+        await hass.async_block_till_done()
+        assert java_mock_config_entry.state == ConfigEntryState.LOADED
+
+    assert await hass.config_entries.async_unload(java_mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert not hass.data.get(DOMAIN)
+    assert java_mock_config_entry.state == ConfigEntryState.NOT_LOADED
+
+
+async def test_setup_entry_failure(
+    hass: HomeAssistant, java_mock_config_entry: MockConfigEntry
+) -> None:
+    """Test failed entry setup."""
+    java_mock_config_entry.add_to_hass(hass)
+
+    with patch(
+        "mcstatus.server.JavaServer.lookup",
+        side_effect=ValueError,
+    ):
+        assert not await hass.config_entries.async_setup(
+            java_mock_config_entry.entry_id
+        )
+
+    await hass.async_block_till_done()
+    assert java_mock_config_entry.state == ConfigEntryState.SETUP_ERROR
+
+
+async def test_setup_entry_not_ready(
+    hass: HomeAssistant, java_mock_config_entry: MockConfigEntry
+) -> None:
+    """Test entry setup not ready."""
+    java_mock_config_entry.add_to_hass(hass)
+
+    with patch(
+        "mcstatus.server.JavaServer.lookup",
+        return_value=JavaServer(host=TEST_HOST, port=TEST_PORT),
+    ), patch(
+        "mcstatus.server.JavaServer.async_status",
+        return_value=OSError,
+    ):
+        assert not await hass.config_entries.async_setup(
+            java_mock_config_entry.entry_id
+        )
+
+    await hass.async_block_till_done()
+    assert java_mock_config_entry.state == ConfigEntryState.SETUP_RETRY
+
+
+async def test_entry_migration(
+    hass: HomeAssistant, v1_mock_config_entry: MockConfigEntry
+) -> None:
     """Test entry migration from version 1 to 3, where host and port is required for the connection to the server."""
-    config_entry_id = create_v1_mock_config_entry(hass)
-    device_entry_id = create_v1_mock_device_entry(hass, config_entry_id)
+    v1_mock_config_entry.add_to_hass(hass)
+
+    device_entry_id = create_v1_mock_device_entry(hass, v1_mock_config_entry.entry_id)
     sensor_entity_id_key_mapping_list = create_v1_mock_sensor_entity_entries(
-        hass, config_entry_id, device_entry_id
+        hass, v1_mock_config_entry.entry_id, device_entry_id
     )
     binary_sensor_entity_id_key_mapping = create_v1_mock_binary_sensor_entity_entry(
-        hass, config_entry_id, device_entry_id
+        hass, v1_mock_config_entry.entry_id, device_entry_id
     )
 
     # Trigger migration.
     with patch(
-        "homeassistant.components.minecraft_server.api.MinecraftServer.__init__",
+        "mcstatus.server.JavaServer.lookup",
```

Comment:
Similar feedback is in #102018 about patch location

Reply:
Waiting for a decision in the linked PR...

Reply:
Done, thanks 👍